bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -564,6 +569,18 @@ public void initialize() {
         latitude = "";
         longitude = "";
 
+        if (config.location.isBlank()) {
+            final PointType userLocation = locationProvider.getLocation();
+            if (userLocation != null) {
+                latitude = String.valueOf(userLocation.getLatitude());
+                longitude = String.valueOf(userLocation.getLongitude());
```

Comment:
I think here you need to set `config.location` to lat/long. If not the value is not used/overwritten at line 590

Reply:
yeah I was still testing that :) You're likely right, as just got back on it :)

Reply:
The logic's been adjusted so it first picks the source, then validates just in case either would give invalid parameters for the API. If successful it will then populate the variables used during API call's.